_posts/2024-07-24-jenkins-agents.md

```diff
@@ -0,0 +1,304 @@
+---
+layout: post
+title:  "[GSoC][Jenkins] Different Jenkins agent configurations and how I tested them for the kworkflow CI"
+date:   2024-07-24
+author: Marcelo Mendes Spessoto Junior
+author_website: https://marcelospessoto.github.io/
+tags: [jenkins, ci, docker, vms]
+---
+
+# Studying Jenkins Agents for the kworkflow CI
+
+## Table of Contents
++ [What is a Jenkins agent?](#what)
++ [Static agents vs. Cloud](#vs)
++ [What I've tried so far](#sofar)
+    + [The Docker Pipeline Plugin (`docker-workflow`)](#dworkflow)
+    + [SSH Agents](#sshagents)
+        + [Configuring the controller as code to implement the agents](#cascode)
+        + [Deploying the agents](#dagents)
+            + [The Docker agent](#dagent)
+            + [The VM agent](#vmagent)
+    + [The Docker Plugin (`docker-plugin`)](#dplugin)
+        + [Configuring Docker as a Cloud provider for kworkflow's Jenkins CI](#cdc)
+        + [Preparing the environment](#pte)
+        + [Creating the Cloud](#ctc)
++ [The integration tests](#tint)
++ [Final thoughts](#ft)
++ [References](#resources)
+
+During my GSoC project of planning a Jenkins CI for kworkflow, I spent plenty of time experimenting with different possibilities of agent types for the infrastructure.
```

Comment:
"of planning a self-hosted Jenkins CI for kworkflow"  (it may be redundant, but I think it is nice to remember readers about the self-hosting part).